x/epochs/keeper/epoch.go

```diff
@@ -20,6 +23,29 @@ func (k Keeper) GetEpochInfo(ctx sdk.Context, identifier string) (types.EpochInf
 	return epoch, true
 }
 
+// AddEpochInfo adds a new epoch info. Will return an error if the epoch fails validation,
+// or re-uses an existing identifier.
+// This method also sets the start time if left unset, and sets the epoch start height.
+func (k Keeper) AddEpochInfo(ctx sdk.Context, epoch types.EpochInfo) error {
```

Comment:
what for? where it's called?

Reply:
You would want to use this instead of SetEpochInfo for additional check at time of setting. For example we can use it in InitGenesis and simplify it by alot. Let's me update it.